keps/sig-node/127-user-namespaces/README.md

```diff
@@ -0,0 +1,437 @@
+  - Score: [8.1 (HIGH) / 8.8 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2021-25741)
+- [CVE-2017-1002101](https://github.com/kubernetes/kubernetes/issues/60813): mitigated, idem
+  - Score: [9.6 (CRITICAL) / 8.8 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2017-1002101)
+- [CVE-2021-30465](https://github.com/opencontainers/runc/security/advisories/GHSA-c3xm-pvg7-gh7r): mitigated, idem
+  - Score: [8.5 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2021-30465)
+- [CVE-2016-8867](https://nvd.nist.gov/vuln/detail/CVE-2016-8867): Privilege escalation inside containers
+  - Score: [7.5 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2016-8867)
+  - https://github.com/moby/moby/issues/27590
+- [CVE-2018-15664](https://nvd.nist.gov/vuln/detail/CVE-2018-15664): TOCTOU race attack that allows to read/write files in the host
+  - Score: [7.5 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2018-15664)
+  - https://github.com/moby/moby/pull/39252
+
+### Goals
+
+Here we use UIDs, but the same applies for GIDs.
+
+- Increase node to pod isolation by mapping user and group IDs
+  inside the container to different IDs in the host. In particular, mapping root
+  inside the container to unprivileged user and group IDs in the node.
+- Increase pod to pod isolation by allowing to use non-overlapping mappings
+  (UIDs/GIDs) whenever possible. IOW, if two containers runs as user X, they run
+  as different UIDs in the node and therefore are more isolated than today.
+  See phase 3 for limitations.
+- Allow pods to have capabilities (e.g. `CAP_SYS_ADMIN`) that are only valid in
+  the pod (not valid in the host).
+- Benefit from the security hardening that user namespaces provide against some
+  of the future unknown runtime and kernel vulnerabilities.
+
+### Non-Goals
+
+- Provide a way to run the kubelet process or container runtimes as an
+  unprivileged process. Although initiatives like
+  [usernetes](https://github.com/rootless-containers/usernetes) and this KEP
+  both make use of user namespaces, it is a different implementation for a
+  different purpose.
+
+## Proposal
+
+This proposal aims to support running pods inside user namespaces. This will
+improve the pod to node isolation (phase 1 and 2) and pod to pod isolation
+(phase 3) we currently have.
+
+This mitigates all the vulnerabilities listed in the motivation section.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want run some pods with privileged capabilities because
+the applications in the pods require it (e.g. `CAP_SYS_ADMIN` to mount a FUSE
+filesystem or `CAP_NET_ADMIN` to setup a VPN) but I don't want this extra
+capability to grant me any extra privileges on the host or other pods.
+
+#### Story 2
+
+As a cluster admin, I want to allow some pods to run in the host user namespace
+if they need a feature only available in such user namespace, such as loading a
+kernel module with `CAP_SYS_MODULE`.
```

Comment:
This story seems confusing.
`CAP_SYS_MODULE` isn't meaningful in user namespaces.

Reply:
I think it is to show a use case that is not possible in user namespaces, so running in the host userns is required

Reply:
Exactly, @AkihiroSuda maybe you missread something? This is saying it is possible to use the host/initial userns (so you can use CAP_SYS_MODULE that inside a userns doesn't do what you need)

Reply:
Sorry overlooked “host” 😓